README.md

```diff
@@ -0,0 +1,186 @@
+# Todo
+
+TodoMVC on Reflect.
+
+## Hacking Locally
+
+The `dev-worker` command runs the worker using [wrangler](https://developers.cloudflare.com/workers/wrangler/).
+
+```bash
+npm install
+
+# start the backend
+npm run dev-worker
+
+# (in another shell) start the frontend
+VITE_WORKER_URL=ws://localhost:8787 npm run dev
+```
+
+## Publishing Worker to Cloudflare
+
+First, get an account at Cloudflare: https://workers.cloudflare.com/.
+
+Then:
+
+```bash
+# publish to Cloudflare
+npx wrangler publish
+
+# run frontend
+VITE_WORKER_URL=wss://<host from previous command> npm run dev
+```
+
+## Authentication and Authorization
+
+Reflect can optionally authenticate users who connect to rooms with your server and authorize their access to the room.
+
+1. Pass some `authToken` to the `Reflect` constructor's `auth` parameter.
+2. Provide `createReflectServer` with an `authHandler` function that authenticates the user and returns whether the user should be allowed in the room.
+
+The signature for the auth handler is as follows:
+
+```ts
+/**
+ * An `AuthHandler` should validate that the user authenticated by `auth` is
+ * authorized to access the room with `roomID`.
+ * @return A promise which resolves to `UserData` for the user if authentication
+ * and authorization is successful, or rejects if authentication or
+ * authorization fail.
+ */
+export type AuthHandler = (auth: string, roomID: string) => Promise<UserData>;
+
+/**
+ * `UserData` must include a `userID` which is unique stable identifier
+ * for the user.
+ * `UserData` has a size limit of 6 KB.
+ * Currently only `userID` is used, but in the future `UserData` may
+ * be passed through to mutators which could use it to supplement
+ * mutator args and to validate the mutation.
+ */
+export type UserData = ReadonlyJSONObject & { userID: string };
+```
+
+### Auth Revalidation
+
+You can invalidate specific users or rooms using the Server Auth API, below.
+
+## Server Auth API
+
+The server has an HTTP API for authentication tasks.
+
+### Auth API Key
+
+All calls to the Auth API must provide an API Key. Configure the API Key using Wrangler:
+
+```bash
+npx wrangler secret put REFLECT_AUTH_API_KEY
+```
+
```

Comment:
and also update the reflect server options/constructor to use REFLECT_AUTH_API_KEY?